### PR TITLE
[SYCL] Introduce sycl-sanitizer tool

### PR DESCRIPTION
**Pull request by alexbatashev**
sycl-sanitizer is a CLI tool to facilitate development of DPC++ applications. Currently sycl-sanitizer is capable of the following diagnostics:
- USM pointer memory leaks. Report is generated at the end of program execution and contains location of leaked pointer malloc call.
- Construction of SYCL buffers from device or shared USM pointer. The tool will abort execution of application if it attempts to create a buffer from device or shared USM pointer.
- Construction of SYCL buffer from host USM pointer of smaller size, than buffer range. The application will be terminated.

Example usage:
```
sycl-sanitizer application.exe --arg1 --arg2
```

### CODE REVIEW

**Comment by alexbatashev** [comment]
Tests: https://github.com/intel/llvm-test-suite/pull/459

**Comment by alexbatashev** [comment]
@smaslov-intel FYI
We can do a similar wrapper for PI trace collector

**Comment by keryell** [comment]
Cool stuff!

File: sycl/tools/sycl-sanitizer/tool.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+int launch(const char *Cmd, const std::vector<const char *> &Args,
+           const std::vector<const char *> &Env) {
+#ifdef _WIN32
+  _spawnve(_P_WAIT, Cmd, const_cast<char *const *>(Args.data()),
+           const_cast<char *const *>(Env.data()));
+  return 0;
+#else
+  return execve(Cmd, const_cast<char *const *>(Args.data()),
+                const_cast<char *const *>(NewEnv.data()));
```
Comment (alexbatashev) [inline sycl/tools/sycl-sanitizer/tool.cpp:31]:
```suggestion
                const_cast<char *const *>(Env.data()));
```

File: sycl/test/abi/sycl_symbols_linux.dump [hunk 2]
```diff
@@ -3922,10 +3959,10 @@ _ZN2cl4sycl7handler20DisableRangeRoundingEv
 _ZN2cl4sycl7handler20associateWithHandlerEPNS0_6detail16AccessorBaseHostENS0_6access6targetE
 _ZN2cl4sycl7handler20setStateSpecConstSetEv
 _ZN2cl4sycl7handler22setHandlerKernelBundleERKSt10shared_ptrINS0_6detail18kernel_bundle_implEE
-_ZN2cl4sycl7handler28setStateExplicitKernelBundleEv
```
Comment (s-kanaev) [inline sycl/test/abi/sycl_symbols_linux.dump:3925]: It is ABI breaking change, isn't it?

Comment (alexbatashev) [reply sycl/test/abi/sycl_symbols_linux.dump:3925]: No, it has been re-ordered

**Comment by alexbatashev** [comment]
replaced with https://github.com/intel/llvm/pull/5389

**Pull request closed without merging**